src/envoy/auth/config.cc

```diff
@@ -13,200 +13,97 @@
-  ENVOY_LOG(debug, "AsyncClientCallbacks [cluster = {}]: {} {}",
-            cluster_->name(), __func__, uri);
-  // Example:
-  // uri  = "https://example.com/certs"
-  // pos  :          ^
-  // pos1 :                     ^
-  // host = "example.com"
-  // path = "/certs"
-  auto pos = uri.find("://");
-  pos = pos == std::string::npos ? 0 : pos + 3;  // Start position of host
-  auto pos1 = uri.find("/", pos);
-  if (pos1 == std::string::npos) pos1 = uri.length();
-  std::string host = uri.substr(pos, pos1 - pos);
-  std::string path = "/" + uri.substr(pos1 + 1);
-
-  MessagePtr message(new RequestMessageImpl());
-  message->headers().insertMethod().value().setReference(
-      Http::Headers::get().MethodValues.Get);
-  message->headers().insertPath().value(path);
-  message->headers().insertHost().value(host);
-
-  request_ = cm_.httpAsyncClientForCluster(cluster_->name())
-                 .send(std::move(message), *this, timeout_);
-}
-
-void AsyncClientCallbacks::Cancel() { request_->cancel(); }
-
-IssuerInfo::IssuerInfo(Json::Object *json) {
-  ENVOY_LOG(debug, "IssuerInfo: {}", __func__);
+// Load issuer config from JSON.
+void LoadIssuerInfo(const Json::Object& json, IssuerInfo* issuer) {
   // Check "name"
-  name_ = json->getString("name", "");
-  if (name_ == "") {
-    ENVOY_LOG(debug, "IssuerInfo: Issuer name missing");
-    failed_ = true;
-    return;
-  }
-  // Check "audience". It will be an empty array if the key "audience" does not
-  // exist
-  try {
-    audiences_ = json->getStringArray("audiences", true);
-  } catch (...) {
-    ENVOY_LOG(debug, "IssuerInfo [name = {}]: Bad audiences", name_);
-    failed_ = true;
-    return;
+  issuer->name = json.getString("name", "");
+  if (issuer->name == "") {
+    throw EnvoyException("Issuer name missing");
   }
+
+  // Check "audience".
+  // It will be an empty array if the key "audience" does not exist.
+  auto audiences = json.getStringArray("audiences", true);
+  issuer->audiences.insert(audiences.begin(), audiences.end());
+
   // Check "pubkey"
-  Json::ObjectSharedPtr json_pubkey;
-  try {
-    json_pubkey = json->getObject("pubkey");
-  } catch (...) {
-    ENVOY_LOG(debug, "IssuerInfo [name = {}]: Public key missing", name_);
-    failed_ = true;
-    return;
-  }
+  auto json_pubkey = json.getObject("pubkey");
+
   // Check "type"
-  std::string pkey_type_str = json_pubkey->getString("type", "");
-  if (pkey_type_str == "pem") {
-    pkey_type_ = Pubkeys::PEM;
-  } else if (pkey_type_str == "jwks") {
-    pkey_type_ = Pubkeys::JWKS;
+  std::string pubkey_type_str = json_pubkey->getString("type", "");
+  if (pubkey_type_str == "pem") {
+    issuer->pubkey_type = Pubkeys::PEM;
+  } else if (pubkey_type_str == "jwks") {
+    issuer->pubkey_type = Pubkeys::JWKS;
   } else {
-    ENVOY_LOG(debug,
-              "IssuerInfo [name = {}]: Public key type missing or invalid",
-              name_);
-    failed_ = true;
-    return;
+    throw EnvoyException(
```

Comment:
We're mostly following Envoy code style here:
https://github.com/envoyproxy/envoy/blob/master/STYLE.md

Exceptions are encouraged, and it will crash (with stack trace) intentionally.